internal/db/p2p.go

```diff
@@ -0,0 +1,223 @@
+
+	err = db.p2p.SetReplicator(ctx, info, collectionNames...)
+	if err != nil {
+		return err
+	}
+
+	return txn.Commit(ctx)
+}
+
+// DeleteReplicator deletes a replicator from the persisted list
+// or specific schemas if they are specified.
+func (db *DB) DeleteReplicator(ctx context.Context, info client.PeerInfo, collectionNames ...string) error {
+	if db.p2p == nil {
+		return errors.New("no p2p system comfigured")
+	}
+	ctx, txn, err := ensureContextTxn(ctx, db, false)
+	if err != nil {
+		return err
+	}
+	defer txn.Discard(ctx)
+
+	err = db.p2p.DeleteReplicator(ctx, info, collectionNames...)
+	if err != nil {
+		return err
+	}
+
+	return txn.Commit(ctx)
+}
+
+// GetAllReplicators returns the full list of replicators with their
+// subscribed schemas.
+func (db *DB) GetAllReplicators(ctx context.Context) ([]client.Replicator, error) {
+	if db.p2p == nil {
+		return nil, errors.New("no p2p system comfigured")
+	}
+	ctx, txn, err := ensureContextTxn(ctx, db, true)
+	if err != nil {
+		return nil, err
+	}
+	defer txn.Discard(ctx)
+	return db.p2p.GetAllReplicators(ctx)
+}
+
+// AddP2PCollections adds the given collections to the P2P system and
+// subscribes to their topics. It will error if any of the provided
+// collection names are invalid.
+func (db *DB) AddP2PCollections(ctx context.Context, collectionNames ...string) error {
+	if db.p2p == nil {
+		return errors.New("no p2p system comfigured")
+	}
+	ctx, txn, err := ensureContextTxn(ctx, db, false)
+	if err != nil {
+		return err
+	}
+	defer txn.Discard(ctx)
+
+	err = db.p2p.AddP2PCollections(ctx, collectionNames...)
+	if err != nil {
+		return err
+	}
+
+	return txn.Commit(ctx)
+}
+
+// RemoveP2PCollections removes the given collections from the P2P system and
+// unsubscribes from their topics. It will error if the provided
+// collection names are invalid.
+func (db *DB) RemoveP2PCollections(ctx context.Context, collectionNames ...string) error {
+	if db.p2p == nil {
+		return errors.New("no p2p system comfigured")
+	}
+	ctx, txn, err := ensureContextTxn(ctx, db, false)
+	if err != nil {
+		return err
+	}
+	defer txn.Discard(ctx)
+
+	err = db.p2p.RemoveP2PCollections(ctx, collectionNames...)
+	if err != nil {
+		return err
+	}
+
+	return txn.Commit(ctx)
+}
+
+// GetAllP2PCollections returns the list of persisted collection names that
+// the P2P system subscribes to.
+func (db *DB) GetAllP2PCollections(ctx context.Context) ([]string, error) {
+	if db.p2p == nil {
+		return nil, errors.New("no p2p system comfigured")
+	}
+	ctx, txn, err := ensureContextTxn(ctx, db, true)
+	if err != nil {
+		return nil, err
+	}
+	defer txn.Discard(ctx)
+
+	return db.p2p.GetAllP2PCollections(ctx)
+}
+
+// AddP2PDocuments adds the given docIDs to the P2P system and
+// subscribes to their topics. It will error if any of the provided
+// docIDs are invalid.
+func (db *DB) AddP2PDocuments(ctx context.Context, docIDs ...string) error {
+	if db.p2p == nil {
+		return errors.New("no p2p system comfigured")
```

Comment:
todo: Move to `errors.go` file.

todo: Fix typo:

```suggestion
		return errors.New("no p2p system configured")
```

Reply:
done